core/src/main/scala/org/apache/spark/memory/StorageMemoryPool.scala

```diff
@@ -73,27 +73,31 @@ class StorageMemoryPool(lock: Object) extends MemoryPool(lock) with Logging {
    *
    * @param blockId the ID of the block we are acquiring storage memory for
    * @param numBytesToAcquire the size of this block
-   * @param numBytesToFree the size of space to be freed through evicting blocks
+   * @param maxNumBytesToFree the maximum amount of space to be freed through evicting blocks
    * @return whether all N bytes were successfully granted.
    */
   def acquireMemory(
       blockId: BlockId,
       numBytesToAcquire: Long,
-      numBytesToFree: Long,
+      maxNumBytesToFree: Long,
       evictedBlocks: mutable.Buffer[(BlockId, BlockStatus)]): Boolean = lock.synchronized {
     assert(numBytesToAcquire >= 0)
-    assert(numBytesToFree >= 0)
+    assert(maxNumBytesToFree >= 0)
     assert(memoryUsed <= poolSize)
-    memoryStore.ensureFreeSpace(blockId, numBytesToFree, evictedBlocks)
-    // Register evicted blocks, if any, with the active task metrics
-    Option(TaskContext.get()).foreach { tc =>
-      val metrics = tc.taskMetrics()
-      val lastUpdatedBlocks = metrics.updatedBlocks.getOrElse(Seq[(BlockId, BlockStatus)]())
-      metrics.updatedBlocks = Some(lastUpdatedBlocks ++ evictedBlocks.toSeq)
+    if (numBytesToAcquire > memoryFree && maxNumBytesToFree > 0) {
+      val additionalMemoryRequired = numBytesToAcquire - memoryFree
+      memoryStore.evictBlocksToFreeSpace(
```

Comment:
`evictBlocksToFreeSpace` already has a form of this check internally. Even if we add the logic here, we can't drop the checks inside of `evictBlocksToFreeSpace` because when storage is evicting storage we need to handle the fact that an RDD cannot evict blocks belonging to the same RDD.

It wouldn't hurt to add the check here, I suppose, since it would save us the hassle of having to iterate over all of the blocks in the memory store.